### PR TITLE
30 categories reducer redux

### LINKED ISSUE #30: Create reducer to populate categories in redux store
We need a categories reducer (i.e. `/reducers/categories.js`) to handle `GET_CATEGORIES_SUCCESS` actions! Use `/reducers/foods.js` as a guide 😊

Don't forget to add this to `combineReducers`!

You will also need to handle the three `GET_CATEGORIES_...` action types in the `info` reducer. Talk to Bryce about how to do this one!

Relates to #29

### PR DESCRIPTION
**Pull request by ajpfoster**
closes #30

### CODE REVIEW

File: client/reducers/categories.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { GET_CATEGORIES_SUCCESS } from './../actions/categories'
+
+export default function categoriesReducer (state = [], action) {
+  switch (action.type) {
+    case GET_CATEGORIES_SUCCESS:
+      return action.categories
+
```
Comment (emilyparkes) [inline client/reducers/categories.js:7]: No new line here please

File: client/reducers/index.js [hunk 2]
```diff
@@ -2,8 +2,10 @@ import { combineReducers } from 'redux'
 
 import foods from './foods'
 import info from './info'
+import categories from './categories'
 
 export default combineReducers({
   foods,
-  info
+  info,
+  categories
```
Comment (emilyparkes) [inline client/reducers/index.js]: Can you order this so it matches the view in the [wiki](https://github.com/matai-2019/foodlist/wiki/Redux-store)

File: client/reducers/info.js [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
+import {
+  GET_CATEGORIES_PENDING,
+  GET_CATEGORIES_SUCCESS,
+  GET_CATEGORIES_ERROR
+} from '../actions/categories'
 
 export default function infoReducer (state = { pending: true }, action) {
   switch (action.type) {
     case GET_FOODS_PENDING:
+    case GET_CATEGORIES_PENDING:
```
Comment (emilyparkes) [inline client/reducers/info.js:17]: Pull from dev as I believe there are more here.

File: tests/client/reducers/categories.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import {
+  getCategoriesSuccess
+} from '../../../client/actions/categories'
+
+import categoriesReducer from '../../../client/reducers/'
+
+describe('reducer', () => {
+  it('getCatergoriesSuccess', () => {
+    // arrange
```
Comment (emilyparkes) [inline tests/client/reducers/categories.test.js:6]: No comments please

File: tests/client/reducers/info.test.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        error: 'ERR: derp'
+      }
+    }
+    const action = getCategoriesPending()
+
+    const newState = infoReducer(currentState, action)
+
+    expect(newState.info.pending).toBe(expected.info.pending)
+    expect(newState.info.error).toBe(expected.info.error)
+  })
```
Comment (emilyparkes) [inline tests/client/reducers/info.test.js:35]: new line separating closing brackets from `it(...)`

File: tests/client/reducers/info.test.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        error: ''
+      }
+    }
+    const action = getCategoriesError('ERR: derp')
+
+    const newState = infoReducer(currentState, action)
+
+    expect(newState.info.error).toBe(expected.info.error)
+  })
+  it('getFoodsSuccess', () => {
```
Comment (emilyparkes) [inline tests/client/reducers/info.test.js]: here too.

File: tests/client/reducers/info.test.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    const currentState = {
+      foods: []
+    }
+    const action = getFoodsSuccess(expected)
+
+    const newState = infoReducer(currentState, action)
+
+    expect(newState.foods).toBe(expected)
+  })
+  it('getFoodsPending', () => {
```
Comment (emilyparkes) [inline tests/client/reducers/info.test.js]: here too.

File: tests/client/reducers/info.test.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+      }
+    }
+    const action = getFoodsPending()
+
+    const newState = infoReducer(currentState, action)
+
+    expect(newState.info.pending).toBe(expected.info.pending)
+    expect(newState.info.error).toBe(expected.info.error)
+  })
+  it('getFoodsError', () => {
```
Comment (emilyparkes) [inline tests/client/reducers/info.test.js]: here too.

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: tests/client/reducers/info.test.js [hunk 9]
```diff
@@ -1,46 +1,155 @@
 import {
-  GET_FOODS_PENDING,
-  GET_FOODS_SUCCESS,
-  GET_FOODS_ERROR } from '../../../client/actions/foods'
+  getCategoriesSuccess,
```
Comment (emilyparkes) [inline tests/client/reducers/info.test.js]: This file has failing tests. Please have passing tests.

**Review by emilyparkes — DISMISSED** [review]

**Review by bryceknz — APPROVED** [review]

**Pull request merged**